pronto/api/signatures/go.py

```diff
@@ -95,3 +102,72 @@ def get_go_terms(accessions):
 def _sort_term(term):
     max_prots = max(s["proteins"] for s in term["signatures"].values())
     return -max_prots, term["id"]
+
+def chunks(iterable: Iterable, n):
+    """Yield chunks of size n from iterable."""
+
+    n = max(1, n)
+    return zip(*[iter(iterable)] * n)
+
+
+def get_go2panther(subfams: list, terms: dict, pg_cur):
+    
+    con = utils.connect_oracle()
+    cur = con.cursor()
+    
+    go_terms = set()
+
+    for subfam_set in chunks(subfams, 1000):
+        cur.execute(
+        """
+            SELECT DISTINCT METHOD_AC, GO_ID
+            FROM INTERPRO.PANTHER2GO
+            WHERE METHOD_AC IN ('{}')
+            """.format("','".join(map(str, subfam_set)))
+        )
+
+        for row in cur:
+            term_id = row[1]
+            pth_fam = row[0].split(':')[0]
+
+            try:
+                term = terms[term_id]
+            except KeyError:
+                term = terms[term_id] = {"id":term_id, "signatures":{}}
+                go_terms.add(term_id)
+
+            try:
+                s = term["signatures"][pth_fam]
+            except KeyError:
+                s = term["signatures"][pth_fam] = [set(), set(), set()]
+
+            s[2].add(row[0])
+
+    cur.close()
+    con.close()
+
+    for terms_chunk in chunks(list(go_terms), 100):
+        go_details = get_go_details(terms_chunk, pg_cur)
+        for go, details in go_details.items():
+            terms[go].update(details)
+    return terms
+
+def get_go_details(term_set: list, pg_cur):
+    go_details = {}
+
+    pg_cur.execute(
+        f"""
+            SELECT t.id, t.name, t.category
+            FROM term t
+            WHERE t.id IN ({','.join('%s' for _ in term_set)})
+            """, term_set
+        )
+    for row in pg_cur:
+        go_id = row[0]
+        go_details[go_id] = {
+            "id":go_id,
+            "name": row[1],
+            "aspect": row[2],
+            }
+
+    return go_details
```

Comment:
PEP-8 & bind variables:

```suggestion
def get_go_details(term_ids: list[str], pg_cur) -> list[dict]:
    details = []

    binds = ["%s" for _ in term_ids]
    pg_cur.execute(
        f"""
        SELECT t.id, t.name, t.category
        FROM term t
        WHERE t.id IN ({','.join(binds)})
        """,
        term_ids
    )

    columns = ("id", "name", "aspect")
    for row in pg_cur.fetchall():
        details.append(dict(zip(columns, row)))

    return details

```